### ISSUE TITLE
Do not work under Debian Buster

### ISSUE THREAD

**Issue by PutinVladimir** [Ximi1970/systray-x#13]
bash SysTray-X-0.1.4-xUbuntu_18.04-amd64-install.sh
```
Installing SysTray-X companion app to:
/home/user/.mozilla/native-messaging-hosts

Installing SysTray-X addon into all profiles listed in:
/home/user/.thunderbird/profiles.ini

Press any key to continue, Ctrl-C to cancel
Installing SysTray-X companion app...
Installing SysTray-X addon...
cp: cannot stat 'systray-x@Ximi1970.xpi': No such file or directory

Addon and companion app installation complete.
App en JSON config have been installed in: /home/user/.mozilla/native-messaging-hosts

Do not forget to install the Thunderbird SysTray-X Add-on
using the standard Thunderbird Add-ons / Extensions store !
```
Ok. I put xpi to /home/user/.thunderbird/user.profile/extensions and activate it into TB.
But what next ?
I don't see tray icon. Minimize TB no tray icon again.
Ok. I guess maybe need start SysTray-X by hand and do this. Now I see tray icon but no any reaction on new messages or clicking on tray icon or minimize TB or close TB(it just closes).
And nothing in debug window.

**Comment by loveless90** [comment]
I would look in the SysTray_X.json file under ~/.mozilla/native-messaging-hosts/, if the variable "path" points to the SysTray-X binary.

Because all i did to make it work, was copying the binary to the directory i like, place the ".json"-file and adjust the "path"-variable and  install the ".xpi"-file via Thunderbird. And maybe restart thunderbird.

**Comment by PutinVladimir** [comment]
> I would look in the SysTray_X.json file under ~/.mozilla/native-messaging-hosts/, if the variable "path" points to the SysTray-X binary.

path is correct
```
$ LANG=c pwd
/home/user/.mozilla/native-messaging-hosts

$ LANG=c ls -1l
total 348
-rwxr-xr-x 1 user user 284912 Apr 13 05:25 SysTray-X
-rw-r--r-- 1 user user    207 Apr 13 14:49 SysTray_X.json
-rw-r--r-- 1 user user  63680 Apr 13 05:25 systray-x@Ximi1970.xpi

$ LANG=c cat SysTray_X.json
{
  "name": "SysTray_X",
  "description": "SysTray-X System Tray App",
  "path": "/home/user/.mozilla/native-messaging-hosts/SysTray-X",
  "type": "stdio",
  "allowed_extensions": [ "systray-x@Ximi1970" ]
}

$ LANG=c ls -1l ~/.thunderbird/user.profile/extensions
total 432
-rw-r--r-- 1 user user  63680 Apr 13 05:25 systray-x@Ximi1970.xpi

```

> Because all i did to make it work, was copying the binary to the directory i like, place the ".json"-file and adjust the "path"-variable and install the ".xpi"-file via Thunderbird. And maybe restart thunderbird.

I guess that problem much deeper. Maybe because TB in debian is apparmored...

**Comment by loveless90** [comment]
Ok, i forgot to say, that the icon was hidden at first (in the pop-menu).

> I guess that problem much deeper. Maybe because TB in debian is apparmored...

If that would be the case, you can still test it with the version from thunderbird.net:
[https://www.thunderbird.net/en-US/thunderbird/all/](url)

That's the version that i use and works fine for me. If it doesn't work, you'll know at least that there are still other problems with your config.

**Comment by PutinVladimir** [comment]
I checked vanilla TB from thunderbird.net.
What I got:
0. When start TB SysTray-X Warning Version mismatch addon and app
1. Icon is added to tray now
2. Clicking on icon minimize\unminimize TB
3. Settings syncing between icon and plugin
4. "Display unread message count" is not working + System notification about new mail not shown
5. Closing TB totally unload it against minimize

TB vanilla version 68.7.0 working(partly 0, 4, 5)
TB debian buster version 68.6.0 not working

**Comment by loveless90** [comment]
What do you mean by:
> "Display unread message count" is not working 

A wrong number (count of unread messages instead of newly received messages; [https://github.com/Ximi1970/systray-x/issues/14)](https://github.com/Ximi1970/systray-x/issues/14) or no number at all.

**Comment by PutinVladimir** [comment]
no number at all.

**Comment by Ximi1970** [comment]
New installers. Please try version 0.1.5

**Comment by PutinVladimir** [comment]
> New installers. Please try version 0.1.5
Installer SysTray-X-0.1.5-Debian_10-amd64-install.sh work without errors now.

But nothing changed.

**Comment by PutinVladimir** [comment]
More investigation.
1. SysTray-X-0.1.5-Debian_10-amd64-install.sh not work anything.
2. systray-x_0.1.5_deb10_amd64.deb partly work.
2.1 depends on gnome-shell-extension-appindicator it is not needed in KDE and working without it but systray-x status broken while not install gnome bullshit.
2.2 Companion start and show icon and share setting between plugin but nothing else work (minimizing TB not minimize to icon, clicking on icon not work, not show new message number).

**Comment by Ximi1970** [comment]
Thanks for checking! :+1: 
It is almost impossible to check all 24 platforms with all possible desktops used.

The DEB package cannot predict what desktop environment you are using, the script installer could make a guess probably. So (for now) the gnome part is always installed. Will put a smarter install / dual GNOME or KDE packageon the whishlist.

**Comment by PutinVladimir** [comment]
> Will put a smarter install / dual GNOME or KDE packageon the whishlist.

Debian has three packages to work with tray:
libappindicator1
libappindicator3-1
gnome-shell-extension-appindicator
and all (or some) of it is need only for very old app. Companion is not one of it. In KDE it work without any of it.
I don't know about gnome or other DE but if there need one of this lib maybe just add it as suggestion not depends at package and write instruction in package description.

**Comment by qwertychouskie** [comment]
Just wanted to pitch in here, yesterday when running normal updates I almost installed the entirety on GNOME Shell on my system, just by clicking "Install Updates".  Currently I have systray-x pinned at 0.1.4 because of this.  IMO `gnome-shell-extension-appindicator` should be a "suggests", not a "depends".  Even "recommends" would be better than "depends, but it would still install GNOME Shell on systems without it unless you manually override it.

Probably better to just suggest that the user installs `gnome-shell-extension-appindicator` if they're running GNOME Shell in README.md.

**Comment by PutinVladimir** [comment]
systray-x_0.2.0_amd64.deb
Nothing changed.

**Comment by Ximi1970** [comment]
As far as I can see there is no dependency on gnome in this package accoording to the build service. Still need to test it for Debian manually, 15 systems to go...

Dependencies for the non-gnome version:

https://build.opensuse.org/package/binary/home:Ximi1970:Mozilla:Add-ons/systray-x/Debian_10/x86_64/systray-x_0.2.0_amd64.deb

Dependencies for the gnome version:

https://build.opensuse.org/package/binary/home:Ximi1970:Mozilla:Add-ons/systray-x-gnome/Debian_10/x86_64/systray-x-gnome_0.2.0_amd64.deb

**Comment by PutinVladimir** [comment]
>As far as I can see there is no dependency on gnome in this package according to the build service.
Dependency is ok now.
But still:
> 2.2 Companion start and show icon and share setting between plugin but nothing else work (minimizing TB not minimize to icon, clicking on icon or menu show\hide not work, not show new message number).

**Comment by Ximi1970** [comment]
@PutinVladimir what desktop are you using? (KDE, GNOME, XFCE, something else?

**Comment by Ximi1970** [comment]
Mixing the script installer package and the repository is also a cause of problems. If your are using the repository package, please make sure there are no systray-x files in:  

~/.thunderbird/xxxxx/extensions/systray-x@Ximi1970
~/.mozilla/native-messaging-hosts/

You can also run the bash installer with:

SysTray-X-xxxxx-Debian_10-amd64-install -u

**Comment by PutinVladimir** [comment]
> @PutinVladimir what desktop are you using? (KDE, GNOME, XFCE, something else?

KDE


> Mixing the script installer package and the repository is also a cause of problems. If your are using the repository package, please make sure there are no systray-x files in:

Yes. I do clean testing every time.
And As I wrote early it is work with vanilla TB, but not with TB from buster repo.

**Comment by Ximi1970** [comment]
I was fixing another bug and bumped into a pure ascii string compare I was using.
Are you using Russian / Cyrillic characters (taking a hint from your user name :smile: )?

If so I need to find out how I am going to test that problem.....

**Comment by PutinVladimir** [comment]
>  Are you using Russian / Cyrillic characters (taking a hint from your user name smile )?

Yes.

```
$ locale
LANG=ru_RU.UTF-8
LANGUAGE=ru:en_US
```

```
$  LANG= apt list --installed thunderbird* lightning*
Listing... Done
lightning-l10n-ru/stable,stable,now 1:68.8.0-1~deb10u1 all [installed]
lightning/stable,stable,now 1:68.8.0-1~deb10u1 all [installed,automatic]
thunderbird-l10n-ru/stable,stable,now 1:68.8.0-1~deb10u1 all [installed]
thunderbird/stable,now 1:68.8.0-1~deb10u1 amd64 [installed,automatic]
```
But vanilla Russian TB https://download.mozilla.org/?product=thunderbird-68.8.0-SSL&os=linux64&lang=ru work fine with same profile.

**Comment by PutinVladimir** [comment]
I was interested what the difference between vanilla and repo. And I did much more investigation and as I suggested before the trouble is Apparmor.

I add to /etc/apparmor.d/local/usr.bin.thunderbird 2 strings and don't forget restart Apparmor (sudo systemctl restart apparmor.service):
```
$ cat /etc/apparmor.d/local/usr.bin.thunderbird
# Site-specific additions and overrides for usr.bin.thunderbird.
# For more details, please see /etc/apparmor.d/local/README.

  # SysTray-X
  @{PROC}/[0-9]*/comm r,
  owner @{HOME}/.mozilla/native-messaging-hosts/* mixr,
```
For global installtion thru deb package. Should be string like this:
```
/usr/bin/SysTray-X mixr,
/usr/lib/mozilla/native-messaging-hosts/* r,

```

And now companion started with TB and action like minimize working.

But I'm not sure about right way for first string.

After this I added few more lines (but everything looks working without it):

`  owner @{HOME}/.config/{kdeglobals,breezerc} r,`
but this line is KDE and breeze theme specific 

`  owner @{HOME}/.cache/icon-cache.kcache wrk,`
again KDE specific

`  @{PROC}/sys/kernel/core_pattern r,`
Don't know why SysTray-X ask this...

Don't see anywhere SysTray-X-0.2.0-Debian_10-amd64-install.sh so tested on 0.1.5.

**Comment by PutinVladimir** [comment]
Need more permissions, but as now work for me work only minimize\unminimize and not shown new messages number don't know how important this.

```
May 21 12:22:37 debian kernel: [ 8643.078117] audit: type=1400 audit(1590038557.185:709): apparmor="DENIED" operation="open" profile="thunderbird" name="/home/user/.cache/qt_compose_cache_little_endian_c04c97609b694403b897effd850db9d7" pid=10823 comm="SysTray-X" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
May 21 12:22:37 debian kernel: [ 8643.086957] audit: type=1400 audit(1590038557.193:710): apparmor="DENIED" operation="open" profile="thunderbird" name="/home/user/.cache/#1318826" pid=10823 comm="SysTray-X" requested_mask="wr" denied_mask="wr" fsuid=1000 ouid=1000
```

**Comment by PutinVladimir** [comment]
Tested 0.2.0 putting SysTray-X to .mozilla/native-messaging-hosts/ and new version systray-x@Ximi1970.xpi to .thunderbird/profile/extensions/ and now everything looks working. Number of new messages also show.

**Comment by Ximi1970** [comment]
Thanks for your research. I will keep in mind the AppArmor settings.  
I also tested Systray-x in a clean Debian 10 KDE but had no problems at all, except for UTF folder names and using them for the count. Working on a fix for that.

**Comment by zcalusic** [comment]
Helo @Ximi1970  et al.

Been following this project for a long time, waiting for it to grow usable and help Thunderbird be great again. :P

Several attempts so far to have it running on Debian failed... Well... 'til yesterday, when I finally detected some leftover files from previous unsuccessful attempts and got rid of them. 0.2.0 deb package now doesn't want to install GNOME to run (XFCE user here, btw), well... finally I got it working, no sweat! I have that much wanted new email counter, which Mozillians killed so many times in the past. One will never understand why they hate it so much... :smile: 

I guess, good job @Ximi1970! Though, I see you also have a donate button now, so I better run and buy you a beer, you definitely deserve one! :+1:

P.S. as somebody already requested in some comment, choosing the font size for the counter would be much appreciated, but I'll see to open another issue for that (currently the number seems too small).
